docs/troubleshooting.md

```diff
@@ -1,6 +1,92 @@
 # Troubleshooting
 
-## dotnet is crashing
+## General steps
+
+If you encounter any issue with OpenTelemetry .NET Automatic Instrumentation,
+there are steps that can help you understand the issue.
+
+### Enable detailed logging
+
+Detailed debug logs can help you troubleshoot instrumentation issues, and can be
+attached to issues in this project to facilitate investigation.
+
+To get the detailed logs from the OpenTelemetry .NET Automatic Instrumentation, set
+the [`OTEL_LOG_LEVEL`](./config.md#internal-logs) environment variable to `debug`
+before the instrumented process starts.
+
+By default, the library writes the log files under predefined [locations](./config.md#internal-logs).
+If needed, change the default location by updating the `OTEL_DOTNET_AUTO_LOG_DIRECTORY`
+environment variable.
+
+After obtaining the logs, remove the `OTEL_LOG_LEVEL`
+environment variable, or set it to less verbose level
+to avoid unnecessary overhead.
+
+### Enable host tracing
+
+[Host tracing](https://github.com/dotnet/runtime/blob/edd23fcb1b350cb1a53fa409200da55e9c33e99e/docs/design/features/host-tracing.md#host-tracing)
+can be used to gather the information needed to investigate the problems
+related to e.g assemblies not being found. Set the following environment variables:
+
+```terminal
+COREHOST_TRACE=1
+COREHOST_TRACEFILE=corehost_verbose_tracing.log
+```
+
+and re-run the application to collect the log.
+
+## Common issues
+
+### Nothing happens
+
+#### Symptoms
+
+There is no telemetry generated.
+There are no logs in OpenTelemetry .NET Automatic Instrumentation internal logs [location](./config.md#internal-logs).
+
+It may occur that the .NET Profiler is unable to attach
+and therefore no logs would be emitted.
+
+#### Solution
+
+The most common reason is that the instrumented application
+has no permissions to load the OpenTelemetry .NET Automatic Instrumentation
+assemblies.
+
+### Performance issues
+
+#### Symptoms
+
+High CPU usage.
```

Comment:
...when running the instrumented service? How do we define high? I know it's tricky, but we might want to provide an example based on some existing issue.

Reply:
Also, any way they can reliably check CPU usage from PowerShell?